homeassistant/components/wled/config_flow.py

```diff
@@ -0,0 +1,125 @@
+"""Config flow to configure the AdGuard Home integration."""
+import logging
+from typing import Any, Dict, Optional
+
+import voluptuous as vol
+from wled import WLED, WLEDConnectionError
+
+from homeassistant.components.wled.const import DOMAIN  # pylint: disable=W0611
+from homeassistant.config_entries import (
+    CONN_CLASS_LOCAL_POLL,
+    SOURCE_ZEROCONF,
+    ConfigFlow,
+)
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME
+from homeassistant.helpers import ConfigType
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WLEDFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a WLED config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by the user."""
+        return await self._handle_config_flow(user_input)
+
+    async def async_step_zeroconf(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle zeroconf discovery."""
+        if user_input is None:
+            return self.async_abort(reason="connection_error")
+
+        # Hostname is format: wled-livingroom.local.
+        host = user_input["hostname"].rstrip(".")
+        name, _ = host.rsplit(".")
```

Comment:
Actually, it is not fragile at all. Since it is a given fact (seems default for Adruino/ESP based stuff, including ESPHome, which does the exact same).

Reply:
Thinking about it a bit more, futures might change.
I'll set a bit more checks around this and add a unit test for it.

Reply:
This is safe - zeroconf validates the hostname field that it matches the RFC6763 style: https://github.com/jstasiak/python-zeroconf/blob/master/zeroconf.py#L217

Reply:
Thanks for clearing that up @OttoWinter 👍